ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -89,7 +90,8 @@ public AlterTableCompactOperation(DDLOperationContext context, AlterTableCompact
     }
 
     //Will directly initiate compaction if an un-partitioned table/a partition is specified in the request
-    if (desc.getPartitionSpec() != null || !table.isPartitioned()) {
+    if (desc.getPartitionSpec() != null || (!table.isPartitioned() && !DDLUtils.isIcebergTable(table) ) || 
```

Comment:
could we simplify 
````
!(table.isPartitioned() || table.getStorageHandler().isPartitioned(table))
````

Reply:
I don't think it can be simplified like that..
Since `org.apache.hadoop.hive.ql.metadata.table.isPartitioned()` will always return false for Iceberg tables, it is not a suitable check for Iceberg, so I am checking `(!table.isPartitioned() && !DDLUtils.isIcebergTable(table))`, which check that the table is unpartitioned and is not Iceberg.

In another part of the condition I am checking if the table is Iceberg and if it is unpartitioned:
`(DDLUtils.isIcebergTable(table) && !table.getStorageHandler().isPartitioned(table))`
I think we have to check if the table is Iceberg because for native tables table.getStorageHandler() will be null.

Reply:
Done